week8/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/worker/SynchronizeServiceWorker.kt

```diff
@@ -0,0 +1,49 @@
+package com.manuflowers.photoinspiration.worker
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.manuflowers.photoinspiration.R
+import com.manuflowers.photoinspiration.application.PhotoInspirationApplication
+import com.manuflowers.photoinspiration.data.PhotosInspirationRepository
+import com.manuflowers.photoinspiration.data.local.database.PhotoInspirationDatabase
+import com.manuflowers.photoinspiration.data.local.database.PhotosDao
+import com.manuflowers.photoinspiration.data.local.preferences.PhotoInspirationPreferences
+import com.manuflowers.photoinspiration.data.remote.networking.RemoteApiManager
+import com.manuflowers.photoinspiration.data.remote.networking.buildApiService
+import com.manuflowers.photoinspiration.util.showNotification
+
+const val NOTIFICATION_CHANNEL_NAME = "Synchronize service channel"
+const val NOTIFICATION_CHANNEL_ID = "Synchronize ID"
+
+class SynchronizeServiceWorker(context: Context, workerParameters: WorkerParameters) :
+    CoroutineWorker(context, workerParameters) {
+
+    private val apiService by lazy { buildApiService() }
+
+    private val remoteApi by lazy { RemoteApiManager(apiService) }
+
+    private val photosDao: PhotosDao =
```

Comment:
You could've exposed some of these functions to build the dao/repo in the App class, and then just call a single function!

Reply:
Done